### ISSUE TITLE
allow optional `(key,val) in arr` syntax for `v-for`

### ISSUE THREAD

**Issue by vprimachenko** [yyx990803/vue#1384]
optionally binding the iteration index (or key when iterating over an object) to a name will simplify work with nested `v-for`s.
```html
<ul v-for="(name,rows) in table | groupByFirst">
	<li v-for="row in rows">{{name}} - {{row}}</li>
</ul>
```
vs
```html
<ul v-for="rows in table | groupByFirst">
	<li v-for="row in rows">{{$parent.$key}} - {{row}}</li>
</ul>
```
i have this feature implemented in my private fork, would submit a pull-request if wanted (or patch as this is few lines only)

**Comment by Mat-Moo** [comment]
+1

**Comment by Morgul** [comment]
+1

@vprimachenko, I'd definitely recommend putting a PR in.

**Comment by JoshZA** [comment]
:+1:

**Comment by vprimachenko** [comment]
as to an PR, how would i proceed, @yyx990803? Contribution guidelines states 
>Checkout a topic branch from dev and merge back against dev.

yet `v-for` lives in `1.0.0-alpha` **and** `1.0.0-beta`. A pull-request on GitHub can only contain a single branch.

also i don't feel comfortable making an public fork just to submit an 10 line patch and abandon later
```diff
diff --git a/src/directives/for.js b/src/directives/for.js
index 31d789c..c3ccf9a 100644
--- a/src/directives/for.js
+++ b/src/directives/for.js
@@ -17 +17,7 @@ module.exports = {
-      this.alias = inMatch[1]
+      var itMatch = inMatch[1].match(/\((.*),(.*)\)/)
+      if (itMatch) {
+        this.iterator = itMatch[1].trim()
+        this.alias = itMatch[2].trim()
+      } else {
+        this.alias = inMatch[1]
+      }
@@ -225,0 +232,3 @@ module.exports = {
+    if (this.iterator) {
+      _.defineReactive(scope, this.iterator, key || index)
+    }
```

**Comment by yyx990803** [comment]
@vprimachenko for now please PR against the beta branch, I'll handle backporting it into the alpha branch.

A 10-line PR is totally fine, people submit 1-line fixes all the time. You should feel comfortable to fork anything in any way you like, nobody's gonna judge you for what you fork ;)

**Comment by yyx990803** [comment]
merged: #1437
